src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/listener/DomainEventsListener.kt

```diff
@@ -0,0 +1,75 @@
+package uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.listener
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter
+import com.fasterxml.jackson.annotation.JsonAnySetter
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import io.awspring.cloud.sqs.annotation.SqsListener
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.service.ReferralService
+
+@Service
+class DomainEventsListener(
+  private val referralService: ReferralService,
+  private val objectMapper: ObjectMapper,
+) {
+
+  private val log = LoggerFactory.getLogger(this::class.java)
+
+  @SqsListener("hmppsdomaineventsqueue", factory = "hmppsQueueContainerFactoryProxy")
```

Comment:
Is this `hmppsdomaineventsqueue` going to be the queue name in all environments? 

Does factory have have an impact on this?